src/Orleans/ApplicationParts/ApplicationPart.cs

```diff
@@ -0,0 +1,13 @@
+namespace Orleans.ApplicationParts
+{
+    /// <summary>
+    /// A part of an Orleans application.
+    /// </summary>
+    public abstract class ApplicationPart
+    {
+        /// <summary>
+        /// Gets the <see cref="ApplicationPart"/> name.
+        /// </summary>
+        public abstract string Name { get; }
```

Comment:
I'm not sure parts need names. I'd be happy to remove this and make it an interface

Reply:
I assume it was for easier troubleshooting. I don't care that much either way, but since it's not bad in itself, I think I would prefer to keep it in the same shape as their infrastructure.